### PR TITLE
Make Ubuntu 18.04 LTS default for PKE on AWS and Azure

### PR DESCRIPTION
**Pull request by Ecsy**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | no
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
- Make Ubuntu 18.04 LTS default for PKE on AWS and Azure.
- Bump Cloudinfo client version
- Replace Calico Kubernetes CNI to Cilium

### Checklist

- [x] Implementation tested (with at least one cloud provider)
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline (TODO)

### CODE REVIEW

File: internal/providers/pke/pkeworkflow/create_cluster.go [hunk 1]
```diff
@@ -22,119 +22,51 @@ import (
...
-			"sa-east-1":      "ami-08d90516f7c661b6b", // South America (Sao Paulo).
-			"us-east-1":      "ami-07079058aa890ee37", // US East (N. Virginia).
-			"us-east-2":      "ami-0faf98ec1c0e28a7e", // US East (Ohio).
-			"us-west-1":      "ami-0bef95b814eae1fc7", // US West (N. California).
-			"us-west-2":      "ami-0ca6e0198325b7be7", // US West (Oregon).
-		}[region], nil
+	if cloudInfoClient != nil {
+		ami, err := cloudInfoClient.PKEImageName("amazon", "pke", "ubuntu", kubeVersion.String(), pkeVersion, region)
+		if err != nil {
+			// fail silently
```
Comment (sagikazarmark) [inline internal/providers/pke/pkeworkflow/create_cluster.go]: Is this a good idea? Should we at least log it? Maybe retry the request?

File: internal/providers/pke/pkeworkflow/create_aws_worker_pool_activity.go [hunk 2]
```diff
@@ -27,20 +27,23 @@ import (
...
 type CreateWorkerPoolActivity struct {
-	clusters       Clusters
-	tokenGenerator TokenGenerator
+	clusters        Clusters
+	tokenGenerator  TokenGenerator
+	cloudInfoClient *cloudinfo.Client
 }
 
-func NewCreateWorkerPoolActivity(clusters Clusters, tokenGenerator TokenGenerator) *CreateWorkerPoolActivity {
+func NewCreateWorkerPoolActivity(clusters Clusters, tokenGenerator TokenGenerator, cloudInfoClient *cloudinfo.Client) *CreateWorkerPoolActivity {
```
Comment (sagikazarmark) [inline internal/providers/pke/pkeworkflow/create_aws_worker_pool_activity.go]: An interface for the client would have been better.

File: internal/providers/pke/pkeworkflow/create_cluster.go (hunk 1 above)
Comment (Ecsy) [reply internal/providers/pke/pkeworkflow/create_cluster.go]: In this case one ami will be chosen from the static map a few lines later.

File: pkg/cloudinfo/services.go [hunk 3]
```diff
@@ -38,3 +41,33 @@ func (c *Client) GetServiceRegions(ctx context.Context, cloudProvider string, se
...
+	if err != nil {
+		return "", errors.WrapIfWithDetails(
+			err, "couldn't get PKE images",
+			"cloudProvider", cloudProvider,
+			"service", service,
+			"region", region,
+			"getImagesOpts", opts,
+		)
+	}
+	if len(images) <= 0 {
```
Comment (siliconbrain) [inline pkg/cloudinfo/services.go]:
```suggestion
	if len(images) == 0 {
```
`len` will never return a negative value.

File: pkg/cloudinfo/services.go [hunk 4]
```diff
@@ -38,3 +41,33 @@ func (c *Client) GetServiceRegions(ctx context.Context, cloudProvider string, se
 
 	return regionIds, nil
 }
+
+func (c *Client) PKEImageName(cloudProvider, service, os, kubeVersion, pkeVersion, region string) (string, error) {
```
Comment (siliconbrain) [inline pkg/cloudinfo/services.go]: Shouldn't this method require a context to be passed by the caller?

File: internal/providers/pke/pkeworkflow/create_cluster.go [hunk 5]
```diff
@@ -25,116 +25,50 @@ import (
...
+		ami, err := pkeImageNameGetter.PKEImageName("amazon", "pke", "ubuntu", kubeVersion.String(), pkeVersion, region)
+		if err != nil {
+			// fail silently
+		}
+		if ami != "" {
+			return ami, nil
+		}
 	}
+
+	// PKE 0.4.19; K8s 1.13.10; OS Ubuntu
```
Comment (orymate) [inline internal/providers/pke/pkeworkflow/create_cluster.go]: you might add a short comment about how/when to update this map

**Pull request merged**